runtime/jsWasmJsSharedMain/src/kotlinx/benchmark/CommonBlackhole.kt

```diff
@@ -0,0 +1,48 @@
+package kotlinx.benchmark
+
+import kotlinx.benchmark.internal.KotlinxBenchmarkRuntimeInternalApi
+
+private const val MAGIC_SIZE: Int = 13
+
+public actual class Blackhole {
+    private val arrayOfAny: Array<Any?> = arrayOfNulls(MAGIC_SIZE)
+    private var currentAnyPosition: Int = 0
+    private fun consumeAny(obj: Any?) {
+        arrayOfAny[currentAnyPosition] = obj
+        currentAnyPosition = if (currentAnyPosition == MAGIC_SIZE - 1) 0 else currentAnyPosition + 1
+    }
+
+    private val arrayOfInt: IntArray = IntArray(MAGIC_SIZE)
+    private var currentIntPosition: Int = 0
+    private fun consumeInt(i: Int) {
+        arrayOfInt[currentIntPosition] = i
+        currentIntPosition = if (currentIntPosition == MAGIC_SIZE - 1) 0 else currentIntPosition + 1
+    }
+
+    internal fun flushMe() {
+        val sums = arrayOfAny.sumOf { it.hashCode() } + arrayOfInt.sum()
+        println("Consumed blackhole value: $sums")
+    }
+
+    actual fun consume(obj: Any?) = consumeAny(obj)
```

Comment:
These functions are actualized as `inline` on native. Do we need the same here?

Reply:
It does make sense to make the functions inline to eliminate a layer of function calls. However, I wouldn't go further and make the `consumeInt/Any` functions inline. V8 should be able to inline them.